## SPDX-FileCopyrightText: 2026 Melissa <theonest262@gmail.com>
## SPDX-License-Identifier: MIT
##
## Determination patch for BSD-3-Clause Hyprland. Retain Hyprland's copyright
## and BSD-3-Clause notice with this patch and its output.

diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index ece6691..a910ee4 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -363,7 +365,7 @@ void CCompositor::initServer(std::string socketName, int socketFd) {
             Debug::log(WARN, "wl_display_add_socket_fd for {} returned {}: skipping", socketName, RETVAL);
     } else {
         // get socket, avoid using 0
-        for (int candidate = 1; candidate <= 32; candidate++) {
+        for (int candidate = envEnabled("DET_HYBRIS") ? 0 : 1; candidate <= 32; candidate++) {
             const auto CANDIDATESTR = ("wayland-" + std::to_string(candidate));
             const auto RETVAL       = wl_display_add_socket(m_wlDisplay, CANDIDATESTR.c_str());
             if (RETVAL >= 0) {
